### PR TITLE
[PYTHON] Communication Identity | Granular Token API

### PR DESCRIPTION
**Pull request by mjafferi-msft**
# Description
**Granular Token Feature (New API version 2023-10-01)**
- Regenerated the REST client from the latest Swagger file
- New scopes added in the CommunicationTokenScope model to be used for token generation.
       - chat.join
       - chat.join.limited
       - voip.join
- Added a new API version CommunicationIdentityServiceVersion.V2023_10_01 that is now the default API version.
- Added tests for testing token generation using various combination of scopes
- Updated recordings for tests
- Updated CHANGELOG & README

Swagger : [Communication Identity Swagger](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/communication/data-plane/Identity/stable/2023-10-01/CommunicationIdentity.json)
Swagger Spec PR : https://github.com/Azure/azure-rest-api-specs/pull/26108

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Review by DominikMe — APPROVED** [review]
lgtm!

File: sdk/communication/azure-communication-identity/CHANGELOG.md
Comment (AikoBB) [inline sdk/communication/azure-communication-identity/CHANGELOG.md:3]: what's the plan for the release? If we are going to ship it before the bot GA then let's make sure the bot identififier deprecation is mentioned here, also we have to get approval for deprecation API changes

File: sdk/communication/azure-communication-identity/README.md
Comment (AikoBB) [inline sdk/communication/azure-communication-identity/README.md:76]: Why are the scopes in lowercase? We’re referring to the enum values of CommunicationTokenScope, so shouldn’t they be in uppercase?

**Review by AikoBB — APPROVED** [review]
looks good to me, just a couple of comments :)

File: sdk/communication/azure-communication-networktraversal/assets.json
Comment (mjafferi-msft) [reply sdk/communication/azure-communication-networktraversal/assets.json:5]: I just updated the ones which failed in the CI build. However, once we release new version, we do have a story to update identity version in dependent SDK's and also update the recordings. So will be done as a part of that story.

File: sdk/communication/azure-communication-identity/CHANGELOG.md
Comment (mjafferi-msft) [reply sdk/communication/azure-communication-identity/CHANGELOG.md:3]: Yes, I believe we would be releasing this before bot GA. So you mean, updating the changelog to mention deprecation of bot identifier changes and getting approval on this PR from Python architects ?

Comment (AikoBB) [reply sdk/communication/azure-communication-identity/CHANGELOG.md:3]: It'll get updated once you will merge to the main branch. Just make sure getting approval from architects for the apiview generated from the main branch which will include also deprecation changes

**Review by xiangyan99 — APPROVED** [review]
LGTM

**Pull request merged**
